adapter.js

```diff
@@ -1,207 +1,82 @@
-
-var pg_host = 'localhost';
-var pg_port = 5432;
-var pg_user = 'afish';
-var pg_password = 'password';
-
-// DO NOT CHANGE UNLESS INSTRUCTED
-var pg_dbName = 'dvdrental';
-
-// Create a Postgres client to setup DVD database
-var client = new PostgresClient();
-var connectionString = 'dbname='+pg_user+' host='+pg_host+' port='+pg_port+' user='+pg_user+' password='+pg_password;
-client.connectSync(connectionString);
-
-// HACK - delete admin realm
-try {
-  require( 'child_process' ).execSync('rm -r realm-object-server/adapter/__admin.realm*');
-}
-catch(e) {}
-
-var args = process.argv;
-args.splice(0,2);
-if (args.length > 0) {
-  for (let index in args) {
-    let val = args[index];
-    if (val == '-reset' ||
-        val == '-r') {
-          console.log(`Dropping replication slot '${pg_dbName}'`);
-          client.exec(`SELECT pg_drop_replication_slot('${pg_dbName}');`);
-          console.log(`Removing ${pg_dbName} database`);
-          client.exec(`DROP DATABASE IF EXISTS ${pg_dbName}`);
-          
-          var deleteFolderRecursive = function(path) {
-            if( fs.existsSync(path) ) {
-              console.log(`Removing ${path}`);
-              fs.readdirSync(path).forEach(function(file,index){
-                var curPath = path + "/" + file;
-                if(fs.lstatSync(curPath).isDirectory()) { // recurse
-                  deleteFolderRecursive(curPath);
-                } else { // delete file
-                  fs.unlinkSync(curPath);
-                }
-              });
-              fs.rmdirSync(path);
-            }
-          };
-          
-          deleteFolderRecursive('./realm-object-server');
-          process.exit()
-    }
-  } 
-}   
-
-process.on('uncaughtException', function(err) {
-  console.log(err);
-});
-
-// Check for dvdrental 
-client.exec(`SELECT EXISTS (SELECT 1 from pg_database WHERE datname='${pg_dbName}')`);
-var existValue = client.getvalue(0, 0);
-if (existValue == 'f') {
-  console.log('Sample Database Not Found!');
-  console.log('--> Creating %s database', pg_dbName);
-  client.exec(`CREATE DATABASE ${pg_dbName};`);
-  
-  // Load the sample data
-  console.log('--> Loading sample data');
-  const
-    spawn = require( 'child_process' ).spawnSync,
-    ls = spawn( 'pg_restore', [ '-U', pg_user, '-d', pg_dbName, './dvdrental.tar' ] );
-  console.log('--> Finished loading sample data\n');
-}
-
-// Drop the rental_id foreign key constraint on inventory table
-client.finish();
 
-// Reconnect to actual database
-connectionString = 'dbname='+pg_dbName+' host='+pg_host+' port='+pg_port+' user='+pg_user+' password='+pg_password;
-client.connectSync(connectionString);
-// Perform check and removal of rental --> inventory constraint
-// Demo is not using rental table so this allows removal of inventory rows
-var foreignKeyCheck = 
```

Comment:
This is super confusing. There is probably a way to remove this along with the corresponding tables by using CASCADE or something like that. But having this code here makes it more difficult to understand what is going n and we should figure out how to get rid of it.